memgpt/data_types.py

```diff
@@ -36,9 +55,8 @@ def __init__(
         model: str,  # model used to make function call
         user: Optional[str] = None,  # optional participant name
         created_at: Optional[str] = None,
-        tool_name: Optional[str] = None,  # name of tool used
```

Comment:
One weird consequence of following this official OpenAI schema is that when you have (in isolation) a message of role "tool" (meaning it was the response message from a function execution), you do not immediately have access to the tool/function "name" anymore (in the previous official OpenAI schema for functions, this existed in the response message itself).

Instead, to get the name you have to do a lookup into the hypothetical "tools" table using the "tool_call_id" to retrieve the tool name.

This means that if our internal agent representation is using the old "functions" style (which it is, and I think it totally fine for now), when we reconstruct the internal agent representation from the database (eg when we load an agent and populate the in-context messages by retrieving from the DB), we will need to do an extra lookup to grab the tool/function name to place inside `message['name']` for the messages with role == 'function':

<img width="569" alt="image" src="https://github.com/cpacker/MemGPT/assets/5475622/7f0b0776-58d8-4935-ab33-ef5ad2cbbf9d">